internal/hardware/validator.go

```diff
@@ -107,7 +110,7 @@ func isNvme(name string) bool {
 // it against a list of predicates. Returns all the reasons the disk
 // was found to be not eligible, or an empty slice if it was found to
 // be eligible
-func (v *validator) DiskIsEligible(ctx context.Context, disk *models.Disk, infraEnv *common.InfraEnv, cluster *common.Cluster, host *models.Host, hostArchitecture string, allDisks []*models.Disk) ([]string, error) {
+func (v *validator) DiskIsEligible(ctx context.Context, disk *models.Disk, infraEnv *common.InfraEnv, cluster *common.Cluster, host *models.Host, inventory *models.Inventory) ([]string, error) {
```

Comment:
host models includes inventory. 
we don't need to import inventory

Reply:
In this context, it may be an inventory that it going to be validated against the host, and then updated on the host later. See https://github.com/openshift/assisted-service/blob/master/internal/host/host.go#L353